compiler/rustc_hir_analysis/src/collect.rs

```diff
@@ -1201,6 +1201,7 @@ fn trait_def(tcx: TyCtxt<'_>, def_id: LocalDefId) -> ty::TraitDef {
 
     let is_marker = tcx.has_attr(def_id, sym::marker);
     let rustc_coinductive = tcx.has_attr(def_id, sym::rustc_coinductive);
+    let is_fundamental = tcx.has_attr(def_id, sym::fundamental);
```

Comment:
can you change the builtin attr to be `only_local` if it isn't already?

Reply:
Nope. This isn't possible without changing the way `AdtKind` serialization works. When we deserialize an ADT, we actually call `has_attr` on it, since we recreate its `AdtFlags` from scratch. Changing this is pretty involved and includes serializing `AdtFlags` directly.